xds/src/test/java/io/grpc/xds/LookasideLbTest.java

```diff
@@ -0,0 +1,156 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+
+/**
+ * Tests for {@link LookasideLb}.
+ */
+public class LookasideLbTest {
+
+  private final Helper helper = mock(Helper.class);
+  private final List<Helper> helpers = new ArrayList<>();
+  private final List<LoadBalancer> balancers = new ArrayList<>();
+  private final LookasideChannelLbFactory lookasideChannelLbFactory =
+      new LookasideChannelLbFactory() {
+        @Override
+        public LoadBalancer newLoadBalancer(
+            Helper helper, AdsStreamCallback adsCallback, String balancerName) {
+          // just return a mock and record helper and balancer.
+          helpers.add(helper);
+          LoadBalancer balancer = mock(LoadBalancer.class);
+          balancers.add(balancer);
+          return balancer;
+        }
+      };
+
+  private LoadBalancer lookasideLb = new LookasideLb(
+      helper, mock(AdsStreamCallback.class), lookasideChannelLbFactory, new LoadBalancerRegistry());
+
+
+  @Test
+  public void handleBalancerNameChanges() throws Exception {
+    assertThat(helpers).isEmpty();
+    assertThat(balancers).isEmpty();
+
+    List<EquivalentAddressGroup> eags = ImmutableList.of();
+    String lbConfigRaw11 = "{\"balancerName\" : \"dns:///balancer1.example.com:8080\"}";
+    @SuppressWarnings("unchecked")
+    Map<String, ?> lbConfig11 = (Map<String, ?>) JsonParser.parse(lbConfigRaw11);
+    ResolvedAddresses resolvedAddresses = ResolvedAddresses.newBuilder()
+        .setAddresses(eags)
+        .setAttributes(Attributes.newBuilder().set(ATTR_LOAD_BALANCING_CONFIG, lbConfig11).build())
+        .build();
+    lookasideLb.handleResolvedAddresses(resolvedAddresses);
+
+    assertThat(helpers).hasSize(1);
+    assertThat(balancers).hasSize(1);
+    Helper helper1 = helpers.get(0);
+    LoadBalancer balancer1 = balancers.get(0);
+    verify(balancer1).handleResolvedAddresses(resolvedAddresses);
+
+    SubchannelPicker picker1 = mock(SubchannelPicker.class);
+    helper1.updateBalancingState(CONNECTING, picker1);
+    verify(helper).updateBalancingState(CONNECTING, picker1);
+
+    String lbConfigRaw12 = "{"
+        + "\"balancerName\" : \"dns:///balancer1.example.com:8080\","
+        + "\"childPolicy\" : [{\"unsupported\" : {\"key\" : \"val\"}}, {\"unsupported_2\" : {}}]"
+        + "}";
+    @SuppressWarnings("unchecked")
+    Map<String, ?> lbConfig12 = (Map<String, ?>) JsonParser.parse(lbConfigRaw12);
+    resolvedAddresses = ResolvedAddresses.newBuilder()
+        .setAddresses(eags)
+        .setAttributes(Attributes.newBuilder().set(ATTR_LOAD_BALANCING_CONFIG, lbConfig12).build())
+        .build();
+    lookasideLb.handleResolvedAddresses(resolvedAddresses);
+
+    verify(balancer1).handleResolvedAddresses(resolvedAddresses);
+
+    verify(balancer1, never()).shutdown();
+    assertThat(helpers).hasSize(1);
+    assertThat(balancers).hasSize(1);
+
+    // change fallback policy to fallback_2
```

Comment:
I guess this comment is wrong.

Reply:
Changed. The test must have copied some other test as a template.